server/lib/nl/detection/place.py

```diff
@@ -37,10 +37,13 @@ def detect_from_query_dc(orig_query: str,
 
   query_items = dc.recognize_places(query)
 
-  nonplace_query_parts = []
-  places_str = []
   mains = []
   main2corrections = {}
+  # Dictionary of dcid to the query substring that matched to the dcid
```

Comment:
I was confused before, and now I see what you're trying to do.  Would it be more direct to do:

```
class QueryPart:
  substr: str
  dcid: str = ''

query_parts: List[QueryPart] = []
```

instead of a list and a map?

Reply:
i like that, updated